docs/concepts/queries.md

```diff
@@ -26,7 +26,7 @@ const api = createApi({
       // `result` is the server response
       onSuccess(id, queryApi, result) {},
       onError(id, queryApi) {},
-      provides: (_, id) => [{ type: 'Post', id }],
+      provides: (result, error, id) => (result ? [{ type: 'Post', id }] : []),
```

Comment:
```suggestion
      provides: (result, error, id) => [{ type: 'Post', id }],
```

we could also think about adding an example like

```ts
      provides: (result, error, id) => (result ? 
        [{ type: 'Post', id }] : 
        (
          'status' in error && error.status === 401) ?
          ['UNAUTHORIZED']:
          ['UNKNOWN_ERROR']
        ),
// in combination with:
      login: build.mutation<unknown, void>({
        query: () => '/login',
        invalidates: ['UNAUTHORIZED'],
      }),
      refetchErroredQueries: build.mutation<unknown, void>({
        queryFn() {
          return { data: {} };
        },
        invalidates: ['UNKNOWN_ERROR'],
      }),

```

maybe even as a recipe and linked in the invalidation chapter?

Reply:
I'm a big fan of this idea. I'll see if I can get something together in a separate PR for that